docs/explanation/lifecycle-layer.rst

```diff
@@ -0,0 +1,198 @@
+Indeed, if we run ``rockcraft prime --shell-after``, we can see the final
+contents ready to be packed in the prime directory - this is the directory
+available at build-time through the ``${CRAFT_PRIME}`` environment variable:
+
+.. code-block:: bash
+
+   $ rockcraft prime --shell-after
+   $ cd ../prime
+   $ ls
+   bin  etc  lib  lib64  sbin  usr  var
+   $ ls usr/bin/
+   debconf               debconf-copydb          debconf-show  dpkg-divert              dpkg-realpath      dpkg-trigger  py3clean     python3
+   debconf-apt-progress  debconf-escape          dpkg          dpkg-maintscript-helper  dpkg-split         perl          py3compile   python3.10
+   debconf-communicate   debconf-set-selections  dpkg-deb      dpkg-query               dpkg-statoverride  perl5.34.0    py3versions  update-alternatives
+
+As we can see, the prime directory has the contents of the ``python3-minimal``
+package but also many of its dependencies, direct and otherwise.
+
+Pruning the ``prime`` directory
+-------------------------------
+
+Once the lifecycle is finished, Rockcraft packs the contents of the prime
+directory as a new OCI layer, directly as if the prime directory was the
+filesystem root ``/``. One consequence of the inclusion of a stage-package's
+dependencies is that the prime directory ends up having many files that the base
+Ubuntu layer already has. This can be seen, for example, by using a tool like
+``dive``:
+
+.. figure:: /_static/dive-efficiency.png
+   :width: 75%
+   :align: center
+   :alt: Rock diagram
+
+What ``dive`` tells us is that about ``60 MB`` worth of files are *duplicated*
+between the base Ubuntu Jammy layer and the "primed" layer: for example, the
+file ``/usr/lib/x86_64-linux-gnu/libcrypto.so.3`` exists both in the base layer
+(as part of the base Ubuntu system) and in the primed layer (pulled in by
+belonging to a package that is an indirect dependency of ``python3-minimal``).
+
+Since version ``1.1.0``, Rockcraft will "prune" those files in the prime
+directory that also exist, with the same contents, ownership and permissions, in
+the base layer. The end-result is semantically the same, because the layers are
+"stacked" together when creating containers from the rock. This "pruning" can be
+seen in the logs generated by Rockcraft:
+
+.. code-block:: text
+
+   (...)
+   Pruning: /root/prime/usr/lib/x86_64-linux-gnu/perl-base/unicore/lib/Sc/Gran.pl as it exists on the base
+   Pruning: /root/prime/usr/lib/x86_64-linux-gnu/perl-base/unicore/lib/Bc/EN.pl as it exists on the base
+   Pruning: /root/prime/usr/lib/x86_64-linux-gnu/perl-base/unicore/lib/PatSyn/Y.pl as it exists on the base
+   Pruning: /root/prime/usr/lib/x86_64-linux-gnu/perl-base/unicore/lib/Dt/Init.pl as it exists on the base
+   Pruning: /root/prime/usr/share/perl5/Debconf/Element/Noninteractive/Multiselect.pm as it exists on the base
+   (...)
+
+
+``usrmerge`` and the lifecycle layer
```

Comment:
This section is really useful. I wonder though if it isn't too in-depth for a regular reader? Unlike the OCI layering (which is a well-known design feature of all container images), this `usrmerge` thing is very specific to the internal workings of the tool. I think only a developer would ever worry about this, no?

Having said that, I don't really have a better place to put it, so I guess it is ok to keep it here unless you have a better idea

Reply:
The original motivation for this page is to document how Rockcraft packs a layer, and with Ubuntu bases there's no way around talking about `usrmerge` because we've had many bug reports and user confusion about this.

This doc isn't meant to be read by everyone, but if someone is having an issue and wants to understand things in more detail then it's useful to have this page to point to.

I actually left out a bunch of extra details, this is the "simplified" version ;)

Reply:
See my comment above in https://github.com/canonical/rockcraft/pull/445#discussion_r1423913874.

In short, putting this side by side with the other existing explanations just feels very unnatural. 

I think this split will become more natural as we move the docs into the already agreed split "Consumer" vs "Developer". But for now, having `usrmerge` explanations at the same level as "What is a rock?" feels like a fruit salad :p